context/app/static/js/components/RecursiveList.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import DataPanel from './DataPanel';
+import RecursiveListLeaf from './RecursiveListLeaf';
+import { isEmptyArrayOrObject } from '../helpers/functions';
+
+const RecursivePropertyContainer = styled.div`
+  margin-top: 5px;
+  font-size: 16px;
+`;
+
+const PropertyName = styled.span`
+  font-weight: bolder;
+  font-size: 1rem;
+`;
+
+function isLeaf(property) {
+  return (typeof property === 'number' || typeof property === 'string' || typeof property === 'boolean' || isEmptyArrayOrObject(property));
```

Comment:
consider using a set?
```suggestion
  return ( new Set(['number', 'string', 'boolean']).has(typeof property) || isEmptyArrayOrObject(property));
```